Source/Adobe.Target.Client/Service/TargetService.cs

```diff
@@ -128,34 +130,14 @@ private void SetUrl(string locationHint)
                 return;
             }
 
-            this.stickyBaseUrl = newUrl;
-            this.SetDeliveryApi();
-        }
-
-        private void SetDeliveryApi()
-        {
-            this.deliveryApi = new DeliveryApi(this.GetDeliveryApiConfig(this.stickyBaseUrl));
+            Interlocked.Exchange(ref this.stickyBaseUrl, newUrl);
```

Comment:
As far as I understand we should atomically update the `DeliveryApi` client `baseUrl`, are we sure `Interocked.Exhange` for URL is enough and we don't need to protect the `DeliveryApi` via a `lock`?

Reply:
Yes, just having the fields volatile and using Interlocked.Exchange is enough, see: https://stackoverflow.com/a/2193445